infra/modules/azure_function_app/tier.tf

```diff
@@ -0,0 +1,37 @@
+## Tier Definitions
+# Centralized mapping for legacy and new tiers
+
+locals {
+  # Tier mappings for better scalability and readability
+  tier_map = {
+    "test"     = "s"
+    "standard" = "m"
+    "premium"  = "l"
+    "xs"       = "xs"
+    "s"        = "s"
+    "m"        = "m"
+    "l"        = "l"
+    "xl"       = "xl"
+  }
+
+  # Default to the mapped tier
+  tier = local.tier_map[var.tier]
+
+  # SKU name mapping
+  sku_name_mapping = {
+    "xs" = "B1"
+    "s"  = "B1"
+    "m"  = "P0v3"
+    "l"  = "P1v3"
+    "xl" = "P2mv3"
+  }
+
+  # Worker process count based on tier
+  worker_process_count_mapping = {
+    "xs" = 2
+    "s"  = 2
+    "m"  = 4
+    "l"  = 8
```

Comment:
cc @BurnedMarshal

Reply:
`xs` and `s` are unstable becouse they have less than 4 GB of memory, set 1 for them anyway.
`m` is P0V3 with 4GB of ram, set 1.
`l` is P1V3 with 8 GB, set 2
`xl`is P2mv3 with 32 GB, set 8

Reply:
@Krusty93 If i've understand what you mean the right table is:

- xs = 1 (because F1 have 1 GB)
- s = 1 (because B1 have 1,75 GB)
- m = 1 (because P0v3 have 3,5 GB)
- l = 1 (because P1v3 have 3,5 GB)
- xl = 2 (because P2v3 have 7 GB)

correct?

Reply:
2 workers for P1v3, as there are 8GBs in there
4 workers for P2V3, as it has 16GB

Reply:
We don't have in functions P2V3, but P2mv3 that has 32 GB of ram, so 8 worker are the right value :D